spec/support/test_setup.rb

```diff
@@ -2,118 +2,54 @@
 class TestSetup
   # A helper class for instantiating a standard set of factories for testing
   #
-  # There are two ways to access the data. Via class methods or class instantiation.
-  #
   # This class automatically generates a set of test data and can be accessed as follows:
   #
-  # e.g. TestSetup.unpublished_with_primary_maths
-  #
-  # Use the above method when you do not need to customise the test data.
-  #
-  # If you do need to custom aspects of the test data then define a 'before(:all)' block in your spec
-  # and instantiate the TestSetup class, passing in the attributes you require:
-  #
-  # e.g @setup = TestSetup.new(with_courses: [
-  #   :unpub_pri_math: { created_date: 2.days.ago }
-  # ])
-  #
-  # [DRAFT] The naming schema for courses is:
-  #
-  #   <publish state>_<level>_<subject>_<other_info>
-  #
-  # where:
+  # e.g
+  # @setup = TestSetup.new
+  # @setup.get_course(:primary, :unpublished)
   #
-  # * publish_state:
+  # Note that a course with the requested traits needs to be pre-defined in this class.
   #
-  #   * pub = published
-  #   * unpub = unpublished
-  #   * del = deleted
-  #
-  # * level:
-  #
-  #   * pri - primary
-  #   * sec - secondary
-  #
-  # * subject:
-  #   * mat - maths
-  #   * eng - eng
+  # It does not matter what order the traits are when passed as arguments.
 
   @@recruitment_cycles = {
-    current_cycle: -> {  find_or_create :recruitment_cycle },
-    next_cycle: -> {  find_or_create :recruitment_cycle, :next },
+      [:current] => -> { FactoryBot.find_or_create :recruitment_cycle },
+      [:next] => -> { FactoryBot.find_or_create :recruitment_cycle, :next },
   }
 
   @@courses = {
-    unpub_pri_math: ->(attributes = {}) do
-      find_or_create(:course, :unpublished_with_primary_maths, attributes)
-    end,
+      [:primary, :unpublished] => ->() do
+        FactoryBot.find_or_create(:course, :primary, :unpublished)
+      end
   }
```

Comment:
Why not make these methods with switches? It doesn't seem like you're passing around the procs here and it might better reflect what you're trying to do.

Reply:
Resolved by renaming the factory collections to make it clearer what they are for :+1: